tests/smoke_tests/smoke_tests_utils.py

```diff
@@ -359,6 +368,28 @@ def run_one_test(test: Test) -> None:
     env_dict = os.environ.copy()
     if test.env:
         env_dict.update(test.env)
+
+    # Create a temporary config file with API server config only if running with remote server
+    if 'PYTEST_SKYPILOT_REMOTE_SERVER_TEST' in os.environ:
+        temp_config = tempfile.NamedTemporaryFile(mode='w',
+                                                  suffix='.yaml',
+                                                  delete=False)
+        if 'SKYPILOT_CONFIG' in env_dict:
+            # Read the original config
+            with open(env_dict['SKYPILOT_CONFIG'], 'r') as f:
+                config = yaml.safe_load(f)
+        else:
+            config = {}
+        host = '0.0.0.0' if is_inside_docker() else 'host.docker.internal'
```

Comment:
Umm, I can't figure out what's going on here, can you elaborate?

Reply:
Apologies for the confusion earlier! Let me fix the Mermaid diagram syntax and provide a valid representation of your explanation.

Here’s the corrected Mermaid diagram:

```mermaid
graph TD
    subgraph "Scenario 2: Local Development (VM/PC)"
        C[VM or Personal PC] -->|Uses 0.0.0.0| D[Container B]
        D -->|Runs Sky Server| D
    end

    subgraph "Scenario 1: Buildkite Tests (Docker-in-Docker)"
        A[Container A] -->|Uses host.docker.internal| B[Container B]
        B -->|Runs Sky Server| B
    end
```

### Explanation:
1. **Scenario 1 (Buildkite Tests)**:
   - Container A communicates with Container B using `host.docker.internal`.
   - Container B runs the Sky Server.

2. **Scenario 2 (Local Development)**:
   - The VM or personal PC communicates with Container B using `0.0.0.0`.
   - Container B runs the Sky Server, and Docker binds the port to `localhost`.

This should now render correctly in any Mermaid-compatible environment. Let me know if you need further adjustments!

Reply:
So looks like the condition should be reverted?
```suggestion
        host = '0.0.0.0' if is_inside_docker() else 'host.docker.internal'
        host = 'host.docker.internal' if is_inside_docker() else '0.0.0.0'
```